libvast/vast/qualified_record_field.hpp

```diff
@@ -0,0 +1,65 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/aliases.hpp"
+#include "vast/detail/operators.hpp"
+#include "vast/fwd.hpp"
+#include "vast/type.hpp"
+
+#include <string>
+#include <tuple>
+#include <vector>
+
+namespace vast {
+
+/// A standalone field of an event type, used to address an index column.
+/// Example: { "zeek.conn.id.orig_h", address_type{} }
+struct qualified_record_field
+  : detail::totally_ordered<qualified_record_field> {
+  qualified_record_field(std::string name_, type type_)
+    : name{std::move(name_)}, type{std::move(type_)} {
```

Comment:
```suggestion
  qualified_record_field(std::string type_name, type field_type)
    : name{std::move(type_name)}, type{std::move(field_type)} {
```

Trailing underscores give the intuition of member variables, which is confusing as argument names.

A proposal: the constructor takes actually the record type name and performs the string concatenation as part of the construction. Then the we can call the `name` member `fully_qualified_name` and everything becomes much clearer.

Reply:
```
qualified_record_field(std::string layout_name, record_field field);
```